src/server/page_server.cpp

```diff
@@ -170,6 +178,51 @@ page_server::connection_handler::receive_disconnect_request (tran_server_conn_t:
   disconnect_thread.detach ();
 }
 
+void
+page_server::connection_handler::abnormal_tran_server_disconnect (css_error_code error_code,
+    bool &abort_further_processing)
+{
+  /* when a transaction server suddenly disconnects, if the page server happens to be be either
+   * proactively sending data or responding to a request, an error is reported from the network layer;
+   * this function is a handler for such an error - see cubcomm::send_queue_error_handler.
+   *
+   * NOTE: if needed, functionality can be extended with more advanced features (ie: retry policy, timeouts ..)
+   * */
+
+  er_log_debug (ARG_FILE_LINE, "abnormal_tran_server_disconnect; request abort futher processing\n");
+  abort_further_processing = true;
+
+  if (!m_abnormal_tran_server_disconnect)
+    {
+      if (m_prior_sender_sink_hook_func)
+	{
+	  // passive transaction server connection
+	  er_log_debug (ARG_FILE_LINE, "abnormal_tran_server_disconnect: PTS disconnected from PS. Error code: %d\n",
+			(int)error_code);
+
+	  {
+	    std::lock_guard<std::mutex> lockg { m_prior_sender_sink_removal_mtx };
+
+	    log_Gl.m_prior_sender.remove_sink (m_prior_sender_sink_hook_func);
+	    m_prior_sender_sink_hook_func = nullptr;
+	  }
+
+	  std::thread disconnect_thread { &page_server::disconnect_tran_server, std::ref (m_ps), this };
```

Comment:
This codes seems identical to the one in else {

Reply:
Fixed